app/controllers/articles_controller.rb

```diff
@@ -1,12 +1,13 @@
 class ArticlesController < ApplicationController
   def index
-    articles=Article.all
+    articles=Article.joins(:author).select('articles.*, authors.name AS author_name') #implementd joins
```

Comment:
don't use custom select unless it's necessary. Add where condition to this and limit the records.
Apply pagination to this using kaminari gem

Reply:
Here i wanted to retrieve all the articles along with their author's name, which is not present in the article table.
Hence i performed a join on article table with author table and selected only author's name from author table, nothing else. Hence i feel using custom select was necessary. 
As this can't be achieved with 'where' because that will include all attributes of author table in the join.
Should I proceed with applying pagination to the same query or you want me to do any changes to this ?

Reply:
you can use eager_load or includes method
```
Article.includes(:author).limit(params[:per_page]) 
articles.as_json(include: [author: {only: [:name]}])
```

Reply:
Changes implemented, please check the latest commit.